components/micro-gateway-core/src/main/ballerina/src/gateway/utils/utils.bal

```diff
@@ -405,6 +405,14 @@ public function printError(string key, string message) {
     log:printError(io:sprintf("[%s] [%s] %s", key, getMessageId(), message));
 }
 
+# Add a full error log with a message and provided key (class) and message ID.
+# + key - The name of the bal file from which the log is printed.
+# + message - The message to be logged.
+# + errorMessage - The error to be logged.
+public function printErrorwithMessage(string key, string message, error errorMessage) {
```

Comment:
Can we use same `printError(string, error)` function with a default-able parameter?

Reply:
Thanks. Fixed in https://github.com/wso2/product-microgateway/pull/973/commits/f7298983c0031cf9e8587e99c49ff7c4885120c4